authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/security/SAMLManager.java

```diff
@@ -0,0 +1,160 @@
+            boolean samlEnabled = samlConfigModel.getEnabled().orElse(false);
+
+            if(samlEnabled) {
+                enableSAML(samlConfigModel);
+            } else {
+                disableSAML();
+            }
+        }
+    }
+
+    private void enableSAML(SAMLConfigModel configModel) {
+        try {
+            RelyingPartyRegistration relyingPartyRegistration = createRegistration(configModel);
+            alertRelyingPartyRegistrationRepository.registerRelyingPartyRegistration(relyingPartyRegistration);
+        } catch(CertificateException | IOException ex) {
+            logger.error("Error enabling saml due to certificate issue.", ex);
+        }
+    }
+
+    private void disableSAML() {
+        alertRelyingPartyRegistrationRepository.unregisterRelyingPartyRegistration();
+    }
+
+    private RelyingPartyRegistration createRegistration(SAMLConfigModel configModel) throws CertificateException, IOException {
+        Optional<String> optionalMetadataUrl = configModel.getMetadataUrl().filter(StringUtils::isNotBlank);
+        RelyingPartyRegistration.Builder builder;
+
+        if (optionalMetadataUrl.isPresent()) {
+            builder = RelyingPartyRegistrations.fromMetadataLocation(optionalMetadataUrl.get()).registrationId("default");
+        } else {
+            String metadataString = filePersistenceUtil.readFromFile(AuthenticationDescriptor.SAML_METADATA_FILE);
+            try (InputStream metadataInputStream = new ByteArrayInputStream(metadataString.getBytes())) {
+                builder = RelyingPartyRegistrations.fromMetadata(metadataInputStream);
+            }
+        }
+
+        if (filePersistenceUtil.uploadFileExists(AuthenticationDescriptor.SAML_SIGNING_CERT_FILE)) {
+            signingCredentialBuilder(builder);
+        }
+        if (filePersistenceUtil.uploadFileExists(AuthenticationDescriptor.SAML_ENCRYPTION_CERT_FILE)) {
+            encryptionCredentialBuilder(builder);
+        }
+        if (filePersistenceUtil.uploadFileExists(AuthenticationDescriptor.SAML_VERIFICATION_CERT_FILE)) {
+            verificationCredentialBuilder(builder);
+        }
+
+        builder.assertingPartyDetails(party -> party.wantAuthnRequestsSigned(configModel.getWantAssertionsSigned().orElse(false)));
+
+        return builder.build();
+    }
+
+    private void signingCredentialBuilder(RelyingPartyRegistration.Builder builder) throws CertificateException, IOException {
+        String signingCertString = filePersistenceUtil.readFromFile(AuthenticationDescriptor.SAML_SIGNING_CERT_FILE);
+        // Get signing private key
+        String signingPrivateKeyString = filePersistenceUtil.readFromFile(AuthenticationDescriptor.SAML_SIGNING_PRIVATE_KEY_FILE);
+        try (InputStream signingPrivateKeyInputStream = new ByteArrayInputStream(signingPrivateKeyString.getBytes())) {
+            RSAPrivateKey signingRSAPrivateKey = RsaKeyConverters.pkcs8().convert(signingPrivateKeyInputStream);
```

Comment:
Verify private key isnt other formats, see if there are other libraries that converts as well